src/Compiler.ts

```diff
@@ -0,0 +1,305 @@
+
+        this.logger.logBuildSuccess(entry.basename, process.hrtime(start));
+        return hasLintError;
+    }
+
+
+    /**
+     * Lints the given file
+     */
+    public async lint (filePath: string, printResults: boolean = true) : Promise<boolean>
+    {
+        return this.lintAll([filePath], printResults);
+    }
+
+
+    /**
+     * Lints the given CSS code.
+     *
+     * Returns whether there were any lint errors
+     */
+    private async lintAll (files: string[], printResults: boolean = true) : Promise<boolean>
+    {
+        let filesToLintMap: UniqueKeyMap = {};
+
+        files.forEach(filePath =>
+        {
+            if (filePath[0] !== "~" && !/\/(node_modules|vendor)\//.test(filePath))
+            {
+                filesToLintMap[filePath] = true;
+            }
+        });
+
+        let filesToLint = Object.keys(filesToLintMap);
+
+        if (filesToLint.length === 0)
+        {
+            return false;
+        }
+
+        let outer = await stylelint.lint({
+            configFile: this.stylelintConfigFile,
+            files: filesToLint,
+            formatter: "string",
+            cache: true,
+            fix: this.options.fix,
+        });
+
+        if (printResults && "" !== outer.output)
+        {
+            this.logger.logToolOutput(`Found ${outer.postcssResults.length} Stylelint issues:`, outer.output);
+        }
+
+        return outer.errored;
+    }
+
+
+    /**
+     * Compiles the code to CSS
+     */
+    private async compileScss (entry: CompilationEntry, fileContent: string): Promise<CompilationResult>
+    {
+        return new Promise(
+            (resolve, reject) =>
+            {
+                sass.render(
+                    {
+                        data: fileContent,
+                        file: entry.src,
+                        outFile: entry.outFilePath,
+                        sourceMap: true,
+                        includePaths: [
+                            path.dirname(entry.src),
+                        ],
+                        outputStyle: "compressed",
+                        importer: (url: string) => this.resolveImport(url),
+                    },
+                    (error: Error|undefined, result: CompilationResult) =>
+                    {
+                        if (error)
+                        {
+                            return reject(error);
+                        }
+
+                        resolve(result);
+                    }
+                );
+            }
+        );
+    }
+
+
+    /**
+     * Handles the post processing
+     */
+    private async postProcess (css: CompilationResult, entry: CompilationEntry) : Promise<PostProcessingResult>
+    {
+        try
+        {
+            return await this.postProcessor.process(css.css, {
+                from: entry.src,
+                to: entry.outFilePath,
+                map: {
+                    annotation: false, //this.options.debug,
```

Comment:
Do we need the commented value or is it old?

Reply:
Yes 😉